### PR TITLE
MINOR: fix corrmap fig titles

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
Weird bug in corrmap where the non-template figures got the template title.

I had a weird local error when testing this tutorial, let's see what Travis says.

### CODE REVIEW

**Comment by agramfort** [comment]
why does it relate to template?

File: mne/preprocessing/ica.py [hunk 1]
```diff
@@ -2231,7 +2231,7 @@ def _find_max_corrs(all_maps, target, threshold):
 
 
 def _plot_corrmap(data, subjs, indices, ch_type, ica, label, show, outlines,
-                  layout, cmap, contours, template=True):
+                  layout, cmap, contours, template=False):
```
Comment (jaeilepp) [inline mne/preprocessing/ica.py]: Looks like ``template`` is now never used.

**Comment by jona-sassenhagen** [comment]
I think the circleci failure is an error ..?

File: mne/preprocessing/ica.py (hunk 1 above)
Comment (jona-sassenhagen) [inline mne/preprocessing/ica.py]: Stupid me.

**Comment by jona-sassenhagen** [comment]
>why does it relate to template?

The function does two kinds of plots. The first is the user-selected template, the second is the selected ICs. We want different labels for the two.

**Comment by Eric89GXL** [comment]
CircleCI log is:

```
Generating gallery for auto_tutorials [  4%] plot_artifacts_correction_ica.py
[Parallel(n_jobs=2)]: Done 366 out of 366 | elapsed:    1.8s finished
WARNING: /home/ubuntu/mne-python/tutorials/plot_artifacts_correction_ica.py failed to execute correctly:Traceback (most recent call last):
  File "/home/ubuntu/mne-python/sphinx-gallery/sphinx_gallery/gen_rst.py", line 450, in execute_code_block
    exec(code_block, example_globals)
  File "<string>", line 12, in <module>
  File "<string>", line 2, in fit
  File "/home/ubuntu/mne-python/mne/utils.py", line 726, in verbose
    return function(*args, **kwargs)
  File "/home/ubuntu/mne-python/mne/preprocessing/ica.py", line 376, in fit
    tstep, reject_by_annotation, verbose)
  File "/home/ubuntu/mne-python/mne/preprocessing/ica.py", line 437, in _fit_raw
    tstep)
  File "/home/ubuntu/mne-python/mne/utils.py", line 533, in _reject_data_segments
    raise RuntimeError('No clean segment found. Please '
RuntimeError: No clean segment found. Please consider updating your rejection thresholds.
```
Looking at the code, looks like you broke the indexing (use index not time)

**Comment by jona-sassenhagen** [comment]
Weird ... on master, the tutorial fails on my Mac on Python 3.6, presumably because ica.fit interprets the int input in the wrong way:

```
Loading data for 284 events and 151 original time points ...
0 bad epochs dropped
fitting ICA from 0 to 13899 seconds
Fitting ICA to data using 305 channels.
Please be patient, this may take some time
Inferring max_pca_components from picks.
    Rejecting  epoch based on MAG : ['MEG 1711']
Artifact detected in [12642, 12943]
    Rejecting  epoch based on MAG : ['MEG 1711']
Artifact detected in [17458, 17759]
Selection by number: 25 components
fitting ICA from 13899 to 27799 seconds
Fitting ICA to data using 305 channels.
Please be patient, this may take some time
Inferring max_pca_components from picks.
Traceback (most recent call last):
  File "tutorials/plot_artifacts_correction_ica.py", line 220, in <module>
    raw, start=start, stop=stop, reject=reject)
  File "<string>", line 2, in fit
  File "/Users/jona/tools/mne-python/mne/utils.py", line 726, in verbose
    return function(*args, **kwargs)
  File "/Users/jona/tools/mne-python/mne/preprocessing/ica.py", line 376, in fit
    tstep, reject_by_annotation, verbose)
  File "/Users/jona/tools/mne-python/mne/preprocessing/ica.py", line 437, in _fit_raw
    tstep)
  File "/Users/jona/tools/mne-python/mne/utils.py", line 533, in _reject_data_segments
    raise RuntimeError('No clean segment found. Please '
RuntimeError: No clean segment found. Please consider updating your rejection thresholds.
```

And with raw.times[-1], I get this:

```
Loading data for 284 events and 151 original time points ...
0 bad epochs dropped
fitting ICA from 0 to 92 seconds
Fitting ICA to data using 305 channels.
Please be patient, this may take some time
Inferring max_pca_components from picks.
    Rejecting  epoch based on MAG : ['MEG 1711']
Artifact detected in [12642, 12943]
Selection by number: 25 components
fitting ICA from 92 to 185 seconds
Fitting ICA to data using 305 channels.
Please be patient, this may take some time
Inferring max_pca_components from picks.
    Rejecting  epoch based on MAG : ['MEG 1711']
Artifact detected in [3612, 3913]
Selection by number: 25 components
fitting ICA from 185 to 277 seconds
Fitting ICA to data using 305 channels.
Please be patient, this may take some time
Inferring max_pca_components from picks.
    Rejecting  epoch based on MAG : ['MEG 1411']
Artifact detected in [12341, 12642]
Selection by number: 25 components
[<ICA  |  raw data decomposition, fit (fastica): 13244 samples, 25 components, channels used: "mag"; "grad">, <ICA  |  raw data decomposition, fit (fastica): 13545 samples, 25 components, channels used: "mag"; "grad">, <ICA  |  raw data decomposition, fit (fastica): 13244 samples, 25 components, channels used: "mag"; "grad">]
```

Hope this fixes it.

**Comment by agramfort** [comment]
good to go?

**Comment by jona-sassenhagen** [comment]
Yes.

**Comment by Eric89GXL** [comment]
@jona-sassenhagen the output you pasted above does not match what CircleCI gives

https://5045-1301584-gh.circle-artifacts.com/0/tmp/circle-artifacts.PKha1H4/html/auto_tutorials/plot_artifacts_correction_ica.html

Can you confirm that you get the same outputs on your system (segments rejected, number of samples used, etc.)?

**Comment by jona-sassenhagen** [comment]
Ah sorry, the output I pasted was what I got *before* the last commit (when the seconds were rounded).

When I run it with this commit on my computer, it looks like what the commit does on circle.

**Comment by jona-sassenhagen** [comment]
```
fitting ICA from 0.0 to 92.56956214990684 seconds
Fitting ICA to data using 305 channels.
Please be patient, this may take some time
Inferring max_pca_components from picks.
    Rejecting  epoch based on MAG : ['MEG 1711']
Artifact detected in [12642, 12943]
Selection by number: 25 components
fitting ICA from 92.56956214990684 to 185.1391242998137 seconds
Fitting ICA to data using 305 channels.
Please be patient, this may take some time
Inferring max_pca_components from picks.
    Rejecting  epoch based on MAG : ['MEG 1711']
Artifact detected in [3612, 3913]
Selection by number: 25 components
fitting ICA from 185.1391242998137 to 277.7086864497205 seconds
Fitting ICA to data using 305 channels.
Please be patient, this may take some time
Inferring max_pca_components from picks.
    Rejecting  epoch based on MAG : ['MEG 1411']
Artifact detected in [12341, 12642]
Selection by number: 25 components
```

**Comment by jona-sassenhagen** [comment]
(This is Py3, Circle seems to be running Py2)

**Comment by Eric89GXL** [comment]
Okay this makes the tutorial work okay. However the API for ICA is currently still broken, i.e. an `int` here *should* work but does not. This is probably a NumPy int problem. Do you have time to look into fixing it that way rather than changing the tutorial?

**Comment by jona-sassenhagen** [comment]
I'd prefer to stick to my other open PRs if possible, as long as they're open ... At least #4289.

**Comment by Eric89GXL** [comment]
Okay I'll open an issue

**Comment by jona-sassenhagen** [comment]
Errrr ... can we still eventually merge the corrmap fig name thing? :)

**Comment by Eric89GXL** [comment]
Is that the `template=False` change? It seems like that should actually be fixed (more explicitly) by changing the function that calls the private function, no?

**Comment by jona-sassenhagen** [comment]
I'm not sure, why? Either template=True or template=False could be a sensible default.

**Comment by Eric89GXL** [comment]
> I'm not sure, why? Either template=True or template=False could be a sensible default.

It's true that either could be a sensible default. My point is rather that somewhere in the code there is a function that calls `_plot_corrmap`, and you are currently trying to change the behavior of that calling function. It's more explicit to make that change in the calling function (with `template=False`) than it is to change the private function.

For example, I don't actually know which function you're trying to change the behavior of -- I'd assume `plot_corrmap` or some related function but I can't know from looking at your changes :)

**Comment by jona-sassenhagen** [comment]
It's funny, cause you wrote that code :D (you refactored it, taking out `_plot_corrmap`).
All of this happens within corrmap. There are two calls to the plotting function, the first for the template, and then iterated calls for the identified maps.

Ok, I see what you mean - but thinking about it, I guess my idea was, this is the *minimal* intervention, cause there are multiple calls with template=False and only one with template=True. So you could argue template=False should have been the default in the first place.

**Comment by Eric89GXL** [comment]
> It's funny, cause you wrote that code :D

My memory for implemented changes is not so good :)

>  I guess my idea was, this is the minimal intervention

Ahh fair enough. I guess we can keep your changes to `ICA.fit` in the tutorial since they don't break anything.

**Comment by Eric89GXL** [comment]
Thanks @jona-sassenhagen

**Comment by jona-sassenhagen** [comment]
Thanks @Eric89GXL ... You can remind (yes, I'm betting on your memory here) me to fix ica.fit when I'm done with #4289.

**Pull request merged**